src/Lecture2.hs

```diff
@@ -70,7 +77,17 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+dropElementFromList :: Int -> [a] -> [a]
+dropElementFromList n list  = take n (fst res) ++ snd res where res = splitAt (n+1) list
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt index list = go 0 list []
+  where go :: Int -> [a] -> [a] -> (Maybe a, [a])
+        go _ [] acc                  = (Nothing, acc)
+        go currentIndex (x : xs) acc = if currentIndex == index
+          then (Just x, acc ++ xs)
+          else go (currentIndex + 1) xs (acc ++ [x])
```

Comment:
It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible. Adding elements to the end of the list is rather slow 🐌